src/nxt_http_static.c

```diff
@@ -235,42 +256,58 @@ nxt_http_static_iterate(nxt_task_t *task, nxt_http_request_t *r,
         nxt_str_set(&chr, "");
     }
 
-    nxt_debug(task, "http static: \"%V\" (chroot: \"%V\")", &shr, &chr);
+    nxt_debug(task, "http static: \"%V\", index: \"%V\" (chroot: \"%V\")",
+              &shr, &idx, &chr);
 #else
-    nxt_debug(task, "http static: \"%V\"", &shr);
+    nxt_debug(task, "http static: \"%V\", index: \"%V\"", &shr, &idx);
 #endif
 #endif /* NXT_DEBUG */
 
-    if (share->is_const) {
+    shr_is_const = share->is_const;
+    idx_is_const = nxt_var_is_const(conf->index);
+
+    if (!shr_is_const || !idx_is_const) {
```

Comment:
What if `chroot` is a variable string?
If you didn't run tests, I suggest running them first.

Reply:
When I wrote that code, I didn't fully understand how `chroot` was being handled.  I had that same question to myself.

See the code without variables in index:

```c
$ grepc -tfd nxt_http_static_iterate
./src/nxt_http_static.c:220:
static void
nxt_http_static_iterate(nxt_task_t *task, nxt_http_request_t *r,
    nxt_http_static_ctx_t *ctx)
{
    ...

    if (share->is_const) {
        nxt_var_raw(share->var, &ctx->share);

#if (NXT_HAVE_OPENAT2)
        if (conf->chroot != NULL && ctx->share_idx == 0) {
            nxt_var_raw(conf->chroot, &ctx->chroot);
        }
#endif

        nxt_http_static_send_ready(task, r, ctx);

    } else {
        ret = nxt_var_query_init(&r->var_query, r, r->mem_pool);
        if (nxt_slow_path(ret != NXT_OK)) {
            nxt_http_request_error(task, r, NXT_HTTP_INTERNAL_SERVER_ERROR);
            return;
        }

        nxt_var_query(task, r->var_query, share->var, &ctx->share);

#if (NXT_HAVE_OPENAT2)
        if (conf->chroot != NULL && ctx->share_idx == 0) {
            nxt_var_query(task, r->var_query, conf->chroot, &ctx->chroot);
        }
#endif

        nxt_var_query_resolve(task, r->var_query, ctx,
                              nxt_http_static_send_ready,
                              nxt_http_static_var_error);
     }
}
```

This is the code before adding variables to index.  It seemed to me that variables in `chroot` were only being handled correctly if `share` also contained variables.  I didn't want to touch code that I didn't understand, and so I left it as is (I think I did; and old pytests pass, so they confirm).

I now checked `test/test_static_chroot.py` and found no tests that set a variable in `chroot` while not using variables in `share`.  We either have tests that have variables in both, or only in `share`.

When I programmed this, I assumed that if `chroot` contained variables, `share` would too (that's what the code was telling me, AFAICR).

So, the question I have now is: is it possible to have variables in `chroot` while not using variables in `share`?  If so, we should probably open a new issue (as it's not completely related to this one; just a little bit), and also provide some tests for that case.

Reply:
>  I didn't fully understand how chroot was being handled. 

It's common for devs, I usually understand them by reading tests to know how to use them, then reading source code.

> It seemed to me that variables in chroot were only being handled correctly if share also contained variables. 

`chroot` option can be a const string or a variable string. Previous logic is we call `nxt_http_static_send_ready()` only both `share` and `chroot` are constant strings.

> I now checked test/test_static_chroot.py and found no tests that set a variable in chroot while not using variables in share. We either have tests that have variables in both, or only in share.

It's better to ask @andrey-zelenkov 

>  is it possible to have variables in chroot while not using variables in share

Both of them can be const string and variables.

Reply:
Okay; I opened a separate issue with that.  I think we should block this PR until we resolve that.